PWGEM/Tasks/gammaConversionsmc.cxx

```diff
@@ -74,188 +76,452 @@ struct GammaConversionsmc {
+    {"hConvPointAbsoluteDistanceRes", "hConvPointAbsoluteDistanceRes", {HistType::kTH1F, {{800, -0.0f, 200.f}}}},
+  };
 
-      {"resolutions/hConvPointRRes", "hConvPointRRes_Rec-MC", {HistType::kTH1F, {{1000, -200.f, 200.f}}}},
-      {"resolutions/hConvPointAbsoluteDistanceRes", "hConvPointAbsoluteDistanceRes", {HistType::kTH1F, {{1000, -0.0f, 200.f}}}},
-    },
+  // Reconstructed info of MC validated V0s histos fV0ReconstructedInfoMCValidatedHistos
+  std::vector<MyHistogramSpec> fV0ReconstructedInfoMCValidatedHistoDefinitions{
+    {"hValidatedPtRec", {HistType::kTH1F, {{800, -0.0f, 25.f}}}},
   };
 
-  enum photonCuts {
-    kPhotonIn = 0,
-    kTrackEta,
-    kTrackPt,
-    kElectronPID,
-    kPionRejLowMom,
-    kPionRejHighMom,
-    kTPCFoundOverFindableCls,
-    kTPCCrossedRowsOverFindableCls,
-    kV0Radius,
-    kArmenteros,
-    kPsiPair,
-    kCosinePA,
-    kPhotonOut
+  // MC info only histos
+  std::vector<MyHistogramSpec> fV0MCInfoOnlyHistoDefinitions{
+    {"hValidatedPtTrue", {HistType::kTH1F, {{800, -0.0f, 25.f}}}},
   };
 
-  std::vector<TString> fPhotCutsLabels{
-    "kPhotonIn",
-    "kTrackEta",
-    "kTrackPt",
-    "kElectronPID",
-    "kPionRejLowMom",
-    "kPionRejHighMom",
-    "kTPCFoundOverFindableCls",
-    "kTPCCrossedRowsOverFindableCls",
-    "kV0Radius",
-    "kArmenteros",
-    "kPsiPair",
-    "kCosinePA",
-    "kPhotonOut"};
+  std::map<std::string, HistPtr> fTrackHistos;
+  std::map<std::string, HistPtr> fV0Histos;
+  std::map<std::string, HistPtr> fV0ResolutionHistos;
+  std::map<std::string, HistPtr> fV0ReconstructedInfoMCValidatedHistos;
+  std::map<std::string, HistPtr> fV0MCInfoOnlyHistos;
+
+  std::string fPrefixReconstructedInfoHistos{"reconstructedInformationOnly/"};
+  std::string fPrefixMCInfoNeededHistos{"MCinformationNeeded/"};
+
+  std::string fFullNameIsPhotonSelectedHisto{fPrefixReconstructedInfoHistos + "v0/afterCuts/IsPhotonSelected"};
+
+  std::map<std::string, size_t> fPhotonCutIndeces{
+    {"kPhotonIn", 0},
+    {"kTrackEta", 1},
+    {"kTrackPt", 2},
+    {"kElectronPID", 3},
+    {"kPionRejLowMom", 4},
+    {"kPionRejHighMom", 5},
+    {"kTPCFoundOverFindableCls", 6},
+    {"kTPCCrossedRowsOverFindableCls", 7},
+    {"kV0Radius", 8},
+    {"kArmenteros", 9},
+    {"kPsiPair", 10},
+    {"kCosinePA", 11},
+    {"kPhotonOut", 12}};
+
+  //~ lXaxis = registry.get<TH1>(HIST("hEvents"))->GetXaxis();
+  //~ lXaxis->SetBinLabel(1, "in");
+  //~ lXaxis->SetBinLabel(2, "int7||sel7");
+  //~ lXaxis->SetBinLabel(3, "cent");
+  //~ lXaxis->SetBinLabel(4, "out");
 
   void init(InitContext const&)
   {
-    TAxis* lXaxis = registry.get<TH1>(HIST("IsPhotonSelected"))->GetXaxis();
-    for (size_t i = 0; i < fPhotCutsLabels.size(); ++i) {
-      lXaxis->SetBinLabel(i + 1, fPhotCutsLabels[i]);
+    for (auto bac : std::vector<std::string>{"beforeCuts/", "afterCuts/"}) {
+
+      // reconstructed data histograms
+      {
+        // track histograms
+        {
```

Comment:
Why are these two extra brackets needed?

Reply:
they are not strictly needed, the outer one could directly be removed, the inner when using different names for lPath. But it is good practice to put code that does not have to speak with each other but is still in the enclosing scope in their own scopes for various reasons.